### DISCUSSION TITLE
Handling ontology versions used in an ARC

### DISCUSSION THREAD

**Discussion by Brilator** [nfdi4plants/nfdi4plants_ontology#71] [General]
Hey all, I'm opening the discussion in this repo as it's about ontologies. However, it's probably more relevant, for tools / implementations. 

To my knowledge, the top-most section (ONTOLOGY SOURCE REFERENCE) of an ARC's `isa.investigation.xlsx` is currently not tackled by any tool. 

The section could look like something this:

ONTOLOGY SOURCE REFERENCE |   |  
-- | -- | --
Term Source Name | PECO | UO
Term Source File | http://purl.obolibrary.org/obo/peco.owl | http://purl.obolibrary.org/obo/uo.owl
Term Source Version | releases/2020-08-21 | releases/2020-03-10
Term Source Description | Plant Experimental Conditions Ontology | Unit ontology

It summarizes the use of different ontologies across the ARC, i.e. within any of the isa.* files (including isa.investigation.xlsx and the isa_study and isa_assay sheets, respectively).

Now, referencing the correct ontology can possibly become tricky (and I didn't try to find out how ISA originally handles this). 
Ontology versions may be diverging
- annotating an ARC via Swate or ARCitect with some time in between, in which ontologies were changed in Swate DB
- reusing an assay from another ARC annotated with another ontology version

**Comment by Brilator** [comment]
One solution might be to use a URL to the ontology term that includes the ontology version 
e.g. not just http://purl.obolibrary.org/obo/EO_0007231

Unfortunately, trying to find an example, I run into a lot of dead links in ontobee and OLS...

**Comment by Brilator** [comment]
@Freymaurer 
@HLWeil 
@JonasLukasczyk 
@kdumschott

**Reply by kdumschott** [reply]
@Hannah-Doerpholz

**Comment by Freymaurer** [comment]
> The section could look like something this:

We could collect all TSR from all term accross all ISA files and auto complete this section. The necessary information is written in SwateDB, BUT i am not sure how to fix dead links and ontology versioning.

Maybe this is the point at which we host our own ontology search + versioning.

**Comment by kdumschott** [comment]
It may be a better idea to reference the github repo containing the ontology file and not just the ontology file itself, since ontology versioning results in dead links. Not sure how this would work automatically though.

**Reply by Brilator** [reply]
This assumes, all ontologies come from the gh repo. 
You might however not use any of the tools for annotation (e.g. do it by hand) and / or ingest data (e.g. assays) from other sources.

**Reply by Brilator** [reply]
Plus, you would still run into version issues. 
Good example from @eik-dahms: what if a taxonomy (Species, Organism) changes? We'd have to update all existing ontology annotations in outdated ARCs.

**Reply by eik-dahms** [reply]
At least for ncbi they seem to track everything they change so you would get something like this: 

https://www.ncbi.nlm.nih.gov/taxonomy/12

Which is fine because you can find the current id.

**Comment by HLWeil** [comment]
I don't think versioning necessarily leads to dead links. Even the contrary might be the case (as Dominik and Eik stated).

In principle, versioned ontologies should have releases whose URIs must never change. By this, pointing to an ontology term in a specific version of an ontology should be permanently working.

The thing is, we could implement this, but how do other ontologies handle this?

**Reply by HLWeil** [reply]
I see one problem here in though:

Currently, in the Annotation Tables of Assay and Study Files, we use full links. This somehow clashes with the whole `Ontology Source References` and versioning concept, as for the mapping we would need to deconstruct the link, which feels a bit counterintuitive.
